Python/ast_unparse.c

```diff
@@ -58,6 +59,33 @@ append_charp(_PyUnicodeWriter *writer, const char *charp)
         } \
     } while (0)
 
+static PyObject *
+replace_inf_value(PyObject *obj)
+{
+    PyObject *inf_str = PyUnicode_FromString("inf");
+    PyObject *int_prefix = PyUnicode_FromString("1e");
+    PyObject *int_part = PyLong_FromLong(1 + DBL_MAX_10_EXP);
+    if (!int_part) {
+        return NULL;
+    }
+    PyObject *int_repr = PyObject_Repr(int_part);
+    Py_DECREF(int_part);
+    if (!int_repr) {
+        return NULL;
+    }
+    PyObject *inf_repr = PyUnicode_Concat(int_prefix, int_repr);
```

Comment:
It may be easier to use `PyUnicode_FromFormat()`.

Also it may be worth to cache these strings.

Reply:
That would be a better way, I'll adress your comments tomorrow. Thanks for reviews